### PR TITLE
Server env test

### PR DESCRIPTION
**Pull request by ajm01**
add testcases for:
- server.env insertion and hover
- bootstrap.properties insertion and hover

### CODE REVIEW

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 1]
```diff
@@ -366,6 +372,68 @@ public static void closeProjectViewTree(RemoteRobot remoteRobot, String appName)
...
+        ProjectFrameFixture projectFrame = remoteRobot.find(ProjectFrameFixture.class, Duration.ofMinutes(2));
+        // get a JTreeFixture reference to the file project viewer entry
+        JTreeFixture projTree = projectFrame.getProjectViewJTree(projectName);
+        projTree.expand(projectName, "src", "main", "liberty", "config");
+        if (!projTree.hasText(fileName)) {
+            // file is not present, nothing to delete
+        } else {
+            // delete either server.env or bootstrap.properties if they were
+            // not part of the original app ie: if the testcases created them.
+            if (deleteServerEnv || deleteBootstrapProps) {
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:429]: I think this logic could lead to a file being deleted that is not marked for deletion. For example, if the method is called with "bootstrap.properties" passed in, and `deleteServerEnv` is true but `deleteBootstrapProps` is false, it looks like "bootstrap.properties" would still get deleted.

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 2]
```diff
@@ -506,13 +587,60 @@ public static void hoverInAppServerXML(RemoteRobot remoteRobot, String hoverTarg
...
+                "const y = %d;" +
+                "java.util.List.of(5, 20, 5, 15).forEach((i)=> {" +
+                "const point = new Point(x + i, y);" +
+                "robot.moveMouse(component, point);})";
+
+        // run the jitter mouse script remotely in the idea
+        editor.runJs(String.format(jitterScript, pointX, pointY));
+    }
+
+    public static void insertEnvCfgInAppCfgFile(RemoteRobot remoteRobot, String projectName, String fileName, String envCfgNameSnippet, String envCfgNameChooserSnippet, String envCfgValueSnippet) {
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:610]: suggest to rename to `insertConfigInAppCfgFile` or `insertConfigInConfigFile`
...and maybe rename the "envCfg" prefixes on the params to "config"

File: src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java [hunk 3]
```diff
@@ -115,12 +120,10 @@ public void testInsertFeatureIntoServerXML() {
             // Replace server.xml content with the original content
             UIBotTestUtils.pasteOnActiveWindow(remoteRobot);
         }
-
-
     }
 
     /**
-     * Tests Liberty Lemminx Extension type ahead support in server.xml for a
+     * Tests liberty-ls type ahead support in server.xml for a
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java:126]: should still be lemminx extension for this LS feature?

File: src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java [hunk 4]
```diff
@@ -164,7 +263,13 @@ public static void prepareEnv(String projectPath, String projectName) {
         projTree.expand(projectName, "src", "main", "liberty", "config");
 
         // open server.xml file
-        UIBotTestUtils.openServerXMLFile(remoteRobot, projectName);
+        UIBotTestUtils.openConfigFile(remoteRobot, projectName, "server.xml");
+
+        // open (or create if need be) server.env file
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java:268]: no longer creating the file correct?

File: src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java [hunk 5]
```diff
@@ -164,7 +263,13 @@ public static void prepareEnv(String projectPath, String projectName) {
...
+        UIBotTestUtils.openConfigFile(remoteRobot, projectName, "server.env");
+
+        // open (or create if need be) bootstrap.properties file
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java:271]: same as above

File: src/test/java/io/openliberty/tools/intellij/it/TestUtils.java [hunk 6]
```diff
@@ -279,6 +279,14 @@ public static void validateStanzaInServerXML(String pathToServerXml, String inse
         }
     }
 
+    public static void validateConfigStringInConfigFile(String pathToConfigFile, String expectedServerEnvString) {
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/TestUtils.java:282]: recommend renaming `expectedServerEnvString` to something like `expectedConfigString`

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 7]
```diff
@@ -477,13 +484,31 @@ public static void closeProjectViewTree(RemoteRobot remoteRobot, String appName)
...
+
     /**
      * Opens a server.xml file for a given app
      *
      * @param remoteRobot The RemoteRobot instance.
-     * @param appName     The string application name
      */
-    public static void openServerXMLFile(RemoteRobot remoteRobot, String appName) {
+    /*
+    public static void openServerXMLFile(RemoteRobot remoteRobot, String appName, String fileName) {
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:511]: do we still need this method around?

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 8]
```diff
@@ -575,34 +601,58 @@ public static void closeAllEditorTabs(RemoteRobot remoteRobot) {
...
+    public static void clickOnFileTab(RemoteRobot remoteRobot, String fileName) {
+        ProjectFrameFixture projectFrame = remoteRobot.find(ProjectFrameFixture.class, Duration.ofSeconds(10));
+
+        try {
+            String xPath = "//div[@accessiblename='" + fileName + "' and @class='SingleHeightLabel']";
+            ComponentFixture actionButton = projectFrame.getActionButton(xPath, "10");
+            actionButton.click();
+
+        } catch (WaitForConditionTimeoutException e) {
+            // server.xml not open, nothing to do
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:620]: update to "file not open..."

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 9]
```diff
@@ -575,34 +601,58 @@ public static void closeAllEditorTabs(RemoteRobot remoteRobot) {
...
     }
 
+    /**
+     * Closes a source file that is open in the editor pane
+     *
+     * @param remoteRobot The RemoteRobot instance.
+     * @param fileName The string file name
+     */
+
+    public static void clickOnFileTab(RemoteRobot remoteRobot, String fileName) {
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:611]: recommend to include "CloseFile" in the name ie. `closeFileWithFileTab` or `clickOnFileTabToClose` 
....unless the method header description is not the correct one

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java (hunk 2 above)
Comment (TrevCraw) [reply src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:697]: These suggestions still apply with the latest changes

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java (hunk 9 above)
Comment (ajm01) [reply src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:611]: updated the header description to accurately reflect what the method is now doing

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**